subprojects/docs/src/docs/userguide/migrating_from_ant.adoc

```diff
@@ -0,0 +1,328 @@
+
+We will start by outlining the kinds of approaches you can take to migrating from Ant and offer some general guidelines.
+
+== General guidelines
+
+When you undertake to migrate a build from Ant to Gradle, you should keep in mind the nature of both what you already have and where you would like to end up. Do you want a Gradle build that mirrors the structure of the existing Ant build? Or do you want to move to something that is more idiomatic to Gradle? This is an important decision that will impact the difficulty of the migration and the benefits you accrue from switching to Gradle.
+
+To understand the implications, it may be useful to look at the two extremes:
+
+ 1. Using <<ant.html#sec:import_ant_build,`ant.importBuild()`>> to import the Ant build wholesale into Gradle.
++
+This approach is quick and simple and works for many Ant-based builds. Your targets become Gradle tasks that you can run, and the dependencies between targets are retained.
++
+The downside is that you lose the advantages of Gradle's conventions, many of its plugins, and many of its important features like incremental build, proper multi-project support, the build cache, and more.
+ 2. Creating the Gradle build from scratch.
++
+This sounds intimidating, but you may be surprised at how much heavy lifting Gradle's plugins can do. You will often find that large swathes of an Ant build are unnecessary in the equivalent Gradle one. The big advantage of this approach is that you gain access to all of Gradle's most important features.
++
+The downside is that many builds will have behavior that can't be replicated by existing plugins, so the migration will take time and effort. You also need a good understanding of what the Ant build is doing in order to match the behavior in Gradle accurately.
+
+With those approaches in mind, the first thing you should decide is whether your project is a good fit for any of the standard Gradle project types. We expect that the vast majority of Ant builds are for JVM-based projects that do the same sorts of things:
+
+ * Compile classes
+ * Run tests
+ * Create JAR packages
+ * Produce Javadoc API documentation
+ * Manage dependencies
+
+There are other common tasks that depend on whether the build produces a library, an application, a webapp, and so on, for which Gradle has specific plugins.
+
+If you have such a build, we recommend that you take the approach of creating a Gradle build from scratch using the appropriate plugins — such as the Java Plugin as described in the chapter <<building_java_projects#building_java_projects,Building Java & JVM projects>> — and configuring them as necessary. You might even want to consider changing the directory structure of your project to match Gradle's conventions, e.g. `src/main/java` for Java source files, although you can usually retain the structure and inform Gradle about it via <<building_java_projects#sec:custom_java_source_set_paths,source-set configuration>>.
```

Comment:
> You might even want to consider changing the directory structure of your project to match Gradle's conventions, e.g. `src/main/java` for Java source files, although you can usually retain the structure and inform Gradle about it via <<building_java_projects#sec:custom_java_source_set_paths,source-set configuration>>.

I think we should not suggest this because I believe changing too many things at once are more likely to result in false starts.

Reply:
Agreed. It works fine for relatively straightforward builds, but they're probably not common. And it could still be done as two steps anyway.